tests/sync/test_moodle.py

```diff
@@ -0,0 +1,298 @@
+                                             patch_get_users_courses, patch_course_get_contents):
+        course_contents: typing.Iterable[pathlib.PurePath] = \
+            list(plugin.list_path(pathlib.PurePath("Wirtschaftsinformatik 2 FS2018")))
+        expected_contents = [
+            pathlib.PurePath('Wirtschaftsinformatik 2 FS2018/02 - Geschäftsprozessmanagement/'
+                             'Geschäftsprozessmanagement/Geschäftsprozessmanagement.pdf'),
+            pathlib.PurePath('Wirtschaftsinformatik 2 FS2018/02 - Geschäftsprozessmanagement/'
+                             'Vielerlei Konzepte im Geschäftsprozessmanagement/index.html'),
+            pathlib.PurePath('Wirtschaftsinformatik 2 FS2018/02 - Geschäftsprozessmanagement/'
+                             'Lösung Aufgabe 1/index.html'),
+            pathlib.PurePath('Wirtschaftsinformatik 2 FS2018/03 - BPMN1 - Introduction to Business Process Management/'
+                             '1. Introduction to Business Process Management/'
+                             '1. Introduction to Business Process Management.pdf'),
+            pathlib.PurePath('Wirtschaftsinformatik 2 FS2018/03 - BPMN1 - Introduction to Business Process Management/'
+                             'Solution Modeling Task 1 - Loan application at Wall Street Oasis (WSO) bank/'
+                             'Modeling Task 1 - Loan application at Wall Street Oasis (WSO) bank.PNG')
+        ]
+        assert course_contents == expected_contents
+
+    @pytest.mark.parametrize('filename', ['foo', 'foo.png', 'foo.html'])
+    def test_create_local_digest(self, plugin, monkeypatch, filename, temppath):
+        testfile = temppath / filename
+        text = 'hello kitovu'
+        testfile.write_text(text)
+
+        mtime = 13371337.4242
+        monkeypatch.setattr(pathlib.Path, 'stat', lambda _self:
+                            FakeStatResult(st_size=len(text), st_mtime=mtime))
+
+        plugin.create_local_digest(testfile)
+        expected_size = 0 if filename == 'foo.html' else len(text)
+
+        assert plugin.create_local_digest(testfile) == f'{expected_size}-13371337'
+
+    def test_create_remote_digest(self, plugin, connect_and_configure_plugin,
+                                  patch_get_users_courses, patch_course_get_contents):
+        course_contents: typing.Iterable[pathlib.PurePath] = list(
+            plugin.list_path(pathlib.PurePath("Wirtschaftsinformatik 2 FS2018")))
+        remote_digests = []
+        check_digests = [
+            '4267895-1520803270',
+            '0-1487838705',
+            '0-1520427130',
+            '2119487-1490374823',
+            '44733-1396277827'
+        ]
+        for item in course_contents:
+            remote_digests.append(plugin.create_remote_digest(item))
+        assert remote_digests == check_digests
+
+    def test_list_path_with_wrong_remote_dir(self, plugin, connect_and_configure_plugin, temppath):
```

Comment:
This test is failing at the moment - it writes a config which isn't used in the test at all, and then we call `list_path` with a path that *does* exist?